WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/MigrationEmailService.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+
+final class MigrationEmailService {
+
+    // MARK: - Dependencies
+
+    private let api: WordPressComRestApi
+
+    // MARK: - Init
+
+    init(api: WordPressComRestApi) {
+        self.api = api
+    }
+
+    convenience init(account: WPAccount) {
+        self.init(api: account.wordPressComRestV2Api)
+    }
+
+    /// Convenience initializer that tries to set the `WordPressComRestApi` property, or fails.
+    convenience init() throws {
+        do {
+            let context = ContextManager.shared.mainContext
+            guard let account = try WPAccount.lookupDefaultWordPressComAccount(in: context) else {
+                throw MigrationError.accountNotFound
+            }
+            self.init(api: account.wordPressComRestV2Api)
+        } catch let error {
+            DDLogError("[\(MigrationError.domain)] Object instantiation failed: \(error)")
+            throw error
+        }
+    }
+
+    // MARK: - Methods
+
+    func sendMigrationEmail() async throws {
+        do {
+            WPAnalytics.track(.migrationEmailTriggered)
+            let response = try await sendMigrationEmail(path: Endpoint.migrationEmail)
+            if !response.success {
+                throw MigrationError.unsuccessfulResponse
+            }
+            WPAnalytics.track(.migrationEmailSent)
+        } catch let error {
+            let properties = trackEventProperties(from: error)
+            WPAnalytics.track(.migrationEmailFailed, properties: properties)
+            DDLogError("[\(MigrationError.domain)] Migration email sending failed: \(error)")
+            throw error
+        }
+    }
+
+    private func sendMigrationEmail(path: String) async throws -> SendMigrationEmailResponse {
+        return try await withCheckedThrowingContinuation { continuation in
+            api.POST(path, parameters: nil) { responseObject, httpResponse in
+                do {
+                    let decoder = JSONDecoder()
+                    let data = try JSONSerialization.data(withJSONObject: responseObject)
+                    let response = try decoder.decode(SendMigrationEmailResponse.self, from: data)
+                    continuation.resume(returning: response)
+                } catch let error {
+                    continuation.resume(throwing: error)
+                }
+            } failure: { error, httpResponse in
+                continuation.resume(throwing: error)
+            }
+        }
+    }
+
+    // MARK: - Track Event Properties
+
+    private func trackEventProperties(from error: Error) -> [String: Any] {
+        let errorType: String
+        switch error {
+        case let error as MigrationError:
+            errorType = error.rawValue
```

Comment:
Non-blocking: Consider providing a `localizedDescription` for `MigrationError`s?

Reply:
That was the first thing I tried! I added a `localizedDescription` to `MigrationError` type but when I called it, the error message was:

<img width="721" alt="CleanShot 2022-11-30 at 21 47 07@2x" src="https://user-images.githubusercontent.com/9609223/204904685-0476096e-eb5a-49d3-b7af-d9c20850a59b.png">

But if I cast the error to `MigrationError`, the `localizedDescription` gets called. Which is still not ideal. Because my initial intention was to call `error.localizedDescription` whether the error is `NSError` or `MigrationError`.

Reply:
Ah, [it might be this](https://forums.swift.org/t/confusing-error-localizeddescription-output/5337/2) @salimbraksa.

Totally fine with me to skip if it's too much trouble.

Reply:
This worked! Thanks @twstokes 🙇 . Resolved in https://github.com/wordpress-mobile/WordPress-iOS/pull/19690/commits/f92d617a06859e857c5c411e55ba56c2e6b64c02